### ISSUE TITLE
Logging objects with circular references

### ISSUE THREAD

**Issue by joshkel** [structured-log/structured-log#54]
I got structured-log installed and working, and so one of the first things I tried to do was log uncaught exceptions in my Angular app.

```
  handleError(error) {
    this.log.error('Error: {@error}', error);
    throw error;
  }
```

It doesn't work, because the `error` object (at least as received by Angular) has circular references.

Would you be open to a PR that either optionally or always uses something like https://github.com/moll/json-stringify-safe within MessageTemplate.toText, to avoid this problem?

**Comment by Wedvich** [comment]
Yep, if you could get a PR in that would be great! I think it should always use the "safe" behavior, as it wouldn't provide any value to opt out of it when that would cause logging to fail anyway.
